samples/compose/src/main/java/com/squareup/sample/compose/launcher/SampleLauncherApp.kt

```diff
@@ -0,0 +1,158 @@
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.gestures.forEachGesture
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.requiredHeight
+import androidx.compose.foundation.layout.requiredWidth
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ListItem
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.darkColors
+import androidx.compose.material.lightColors
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.input.pointer.PointerEventPass.Initial
+import androidx.compose.ui.input.pointer.consumeDownChange
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.layout.LayoutCoordinates
+import androidx.compose.ui.layout.boundsInRoot
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.node.Ref
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.core.app.ActivityOptionsCompat.makeScaleUpAnimation
+import androidx.core.content.ContextCompat.startActivity
+import com.squareup.sample.compose.R
+
+@Composable fun SampleLauncherApp() {
+  MaterialTheme(colors = darkColors()) {
+    Scaffold(
+        topBar = {
+          TopAppBar(title = {
+            Text(stringResource(R.string.app_name))
+          })
+        }
+    ) {
+      LazyColumn {
+        items(samples) {
+          SampleItem(it)
+        }
+      }
+    }
+  }
+}
+
+@Preview @Composable private fun SampleLauncherAppPreview() {
+  SampleLauncherApp()
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable private fun SampleItem(sample: Sample) {
+  val rootView = LocalView.current
+
+  /**
+   * [androidx.compose.ui.layout.LayoutCoordinates.globalBounds] corresponds to the coordinates in
+   * the root Android view hosting the composition.
+   */
+  val globalBounds = remember { Ref<Rect>() }
+
+  ListItem(
+      text = { Text(sample.name) },
+      secondaryText = { Text(sample.description) },
+      singleLineSecondaryText = false,
+      // Animate the activities as scaling up from where the preview is drawn.
```

Comment:
Fwiw, I keep find myself reverting to the idea of `@Preview` whenever I read 'preview' here and below. (Admittedly, the functions currently called by this code path are indeed annotated with `@Preview`, but looks like that's coincidental and not inevitable?) Might be worth another term if this one isn't already too baked into this repo's samples. (Nicely, 'sample' might even work as a substitute.)

Reply:
The `preview` function is intended to only/primarily be used inside `@Preview` functions, yes.